lib/arrangement.js

```diff
@@ -17,11 +17,10 @@ module.exports = class Arrangement {
       throw new ArrangementInvalidError(`Invalid ${digest}`)
     }
     if (data.version < 3 || !data.data.b || !data.data.b.length) {
-      kinesis.putMissingDigest(digest)
       throw new ArrangementNoBytesError(`Old ${digest}`)
     }
     if (data.version < 4 || !data.data.a || data.data.a.length !== 3) {
-      kinesis.putMissingDigest(digest) // allow, but queue for upgrade
+      log.warn('Non v4 arrangement', {digest}) // allow, but warn
```

Comment:
_In theory_ we should be past having to deal with pre dovetail-stitch-lambda arrangements.  Anything served in the past ~6 months will already be upgraded to v4 as it's stitched.  (Before the counts-lambda is invoked).

I've been watching that "upgrade arrangements" stream, and traffic has dropped way off to just a few a day.  Anything still coming through it is very old - like arrangements that haven't been seen in more > 18 months.  And I'm willing to lose a few impressions off those.  (The downloads will still be counted though, since we do know the overall length of the file).